bin/dotd

```diff
@@ -109,118 +163,122 @@ def content_push(server_name, environment_name, user_name)
+    # Truncate the message to its first line.
+    commit_message_truncated = "    #{commit_message[/.*/].chomp}"
+    puts dim(commit_message_truncated).to_s
   end
+  puts ''
 end
 
-puts <<-EOS.unindent
-
-  #{bold 'Welcome to Robo-Dev-of-the-Day!'}
-
-    d[ o_0 ]b
-
-  #{bold 'Documentation'}
-
-    Dev-of-the-Day Checklist
-    #{dim 'http://wiki.code.org/display/PROD/Dev+of+the+Day+Checklist'}
+# Prints the DTT subtopic, formatted based on its contents.
+def print_dtt_subtopic
+  if DevelopersTopic.dtt?
+    output_text = dim "CURRENT DTT SUBTOPIC: #{DevelopersTopic.dtt_subtopic}"
+  else
+    output_text = red "CURRENT DTT SUBTOPIC: #{DevelopersTopic.dtt_subtopic}"
+  end
+  puts output_text
+end
 
-    Known Issues
-    #{dim 'http://wiki.code.org/display/PROD/Known+Issues'}
+# Asks if the DTT subroutine should be performed, doing so if requested.
+def do_dtt_subroutine
+  print_dtt_subtopic
+  should_i "DTT" do
+    view_commits base: 'test', head: 'staging'
+    should_i "merge from staging to test" do
+      DevelopersTopic.set_dtt_subtopic "no (@#{@dotd_name} in progress)"
+      pr_number = GitHub.create_and_merge_pull_request(
+        base: 'test',
+        head: 'staging',
+        title: 'DTT (Staging > Test)'
+      )
+      new_sha = GitHub.sha 'test'
+      url = GitHub.html_url(pr_number)
+      wait_for "DTT (#{url}) to complete, re-run / investigate failures, and "\
+        "update infra#test topic to commit #{new_sha}"
+      # TODO(asher): Prompt for whether the DTT was green, update
+      # Slack#infra-test topic appropriately.
+
+      should_i "reset DTT subtopic to yes", log_action: false do
+        DevelopersTopic.set_dtt_subtopic "yes"
+      end
+    end
+  end
+end
 
-EOS
+def main
+  puts_script_intro
+  ask_for_name
 
-user_name = ask_for_name
+  do_dtt_subroutine
 
-should_i "morning DTT" do
-  view_commits base: 'test', head: 'staging'
-  should_i "merge from staging to test" do
-    pr_number = GitHub.create_and_merge_pull_request(
-      base: 'test',
-      head: 'staging',
-      title: 'DTT (Staging > Test)'
-    )
-    new_sha = GitHub.sha 'test'
-    url = GitHub.html_url(pr_number)
-    wait_for "DTT (#{url}) to complete, re-run / investigate failures, and update infra#test topic to commit #{new_sha}"
+  should_i "commit content changes from staging" do
+    content_push 'staging', 'staging'
   end
-end
 
-should_i "commit content changes from staging" do
-  content_push 'staging', 'staging', user_name
-end
+  should_i "commit content changes from levelbuilder" do
+    content_push 'levelbuilder-staging', 'levelbuilder'
+
+    view_commits base: 'staging', head: 'levelbuilder'
+    should_i "merge from levelbuilder to staging" do
+      GitHub.create_and_merge_pull_request(
+        base: 'staging', head: 'levelbuilder', title: 'DTS (Levelbuilder > Staging)'
+      )
+      new_sha = GitHub.sha 'staging'
+      new_sha_abbr = new_sha[0, 4] + '...' + new_sha[-4, 4]
```

Comment:
The git/GitHub convention for abbreviating commit SHAs is `new_sha[0, 7]` as it's almost always unique - I think that'd be sufficient here.